server/src/test/java/org/opensearch/search/pipeline/SearchPipelineServiceTests.java

```diff
@@ -96,14 +99,31 @@ public Map<String, Processor.Factory<SearchPhaseResultsProcessor>> getSearchPhas
 
     private ThreadPool threadPool;
 
+    private IndexNameExpressionResolver indexNameExpressionResolver;
+
     @Before
     public void setup() {
         threadPool = mock(ThreadPool.class);
         ExecutorService executorService = OpenSearchExecutors.newDirectExecutorService();
+        indexNameExpressionResolver = new IndexNameExpressionResolver(new ThreadContext(Settings.EMPTY));
         when(threadPool.generic()).thenReturn(executorService);
         when(threadPool.executor(anyString())).thenReturn(executorService);
     }
 
+    private static IndexMetadata.Builder indexBuilder(String index) {
+        return indexBuilder(index, Settings.EMPTY);
+    }
+
+    private static IndexMetadata.Builder indexBuilder(String index, Settings additionalSettings) {
+        return IndexMetadata.builder(index).settings(settings(additionalSettings));
+    }
+
+    private static Settings.Builder settings(Settings additionalSettings) {
```

Comment:
This seems to do a bit more internally with the settings than the method name/argument implies.

Reply:
Changed it to `addAdditionalSettings`